### PR TITLE
Milestone/0.7.0

### PR DESCRIPTION
**Pull request by dchertousov**
@shmuga 
Her you have a lot of work to do :)
@articice  have a look too, please

### CODE REVIEW

**Comment by dchertousov** [comment]
@shmuga, is it works for you, proceed with merge?

File: src/Services/Air/AirValidator.js [hunk 1]
```diff
@@ -208,6 +208,39 @@ Validator.prototype.flightInfoItem = function (item) {
...
+    Object.prototype.toString.call(this.params.fop) !== '[object Object]' ||
+    this.params.fop.type !== 'Cash'
+  ) {
+    throw new AirValidationError.FopTypeUnsupported();
+  }
+  return this;
+};
+
+Validator.prototype.ticketNumber = function () {
+  if (!this.params.ticketNumber) {
```
Comment (shmuga) [inline src/Services/Air/AirValidator.js]: Will pass if ticketNumber is `Number` and fail later on .match call

File: src/Services/Air/AirValidator.js [hunk 2]
```diff
@@ -208,6 +208,39 @@ Validator.prototype.flightInfoItem = function (item) {
...
+    throw new AirValidationError.TicketNumberMissing();
+  }
+  if (!this.params.ticketNumber.match(/^\d{13}/)) {
+    throw new AirValidationError.TicketNumberInvalid();
+  }
+  return this;
+};
+
+Validator.prototype.paramsIsObject = function () {
```
Comment (shmuga) [inline src/Services/Air/AirValidator.js]: I think it's better to move it to `uapi-request.js` checks. How do you think?

File: src/Services/Air/AirErrors.js [hunk 3]
```diff
@@ -70,32 +74,39 @@ export const AirRuntimeError = createErrorClass(
...
-  'Air flight info service runtime error',
-  errorTypes.RuntimeError
-);
-
-export const GdsRuntimeError = createErrorClass(
-  'GdsRuntimeError',
-  'Gds service runtime error',
-  errorTypes.RuntimeError
-);
 Object.assign(AirRuntimeError, createErrorsList({
```
Comment (dchertousov) [inline src/Services/Air/AirErrors.js]: Just moved in order to keep extending base class right after its declaration

File: src/Services/Air/AirParser.js [hunk 4]
```diff
@@ -435,6 +434,89 @@ const AirErrorHandler = function (obj) {
...
+    }
+  );
+  const taxes = Object.keys(airPricingInfo['air:TaxInfo']).map(
+    taxKey => ({
+      type: airPricingInfo['air:TaxInfo'][taxKey].Category,
+      value: airPricingInfo['air:TaxInfo'][taxKey].Amount,
+    })
+  );
+  const response = {
+    type: 'airTicketDocument',
```
Comment (dchertousov) [inline src/Services/Air/AirParser.js]: Well, it's not necessary.

File: src/Services/Air/AirValidator.js (hunk 2 above)
Comment (dchertousov) [inline src/Services/Air/AirValidator.js]: Not really, as it checks just input object.
For example, `TerminalService` method `executeCommand` requires just `String`, not an object.

**Pull request merged**